README.md

```diff
@@ -45,17 +45,23 @@ For example:
 
 ```json
 {
-  "name": "foo-app",
-  "env": {
-      "BACKGROUND_COLOR": {
-          "description": "specify a css color",
-          "value": "#fefefe",
-          "required": false
-      },
-      "TITLE": {
-          "description": "title for your site"
-      }
-  }
+    "name": "foo-app",
+    "env": {
+        "BACKGROUND_COLOR": {
+            "description": "specify a css color",
+            "value": "#fefefe",
+            "required": false
+        },
+        "TITLE": {
+            "description": "title for your site"
+        },
+        "APP_SECRET": {
+            "generator": "secret"
+        }
+    },
+    "hook": {
+        "postcreate": "postcreate.sh"
```

Comment:
# option1

- "./postcreate.sh --install" -> shlex (favored, for now)

- ["./postcreate.sh", "--install"]

execve($0)

# option2 

/bin/sh -c "$0"

```
"postcreate": "date\ninstall_foo\ninstall_bar" --> `;` vs `&&`
```

# single entrypoint

```
#!/usr/bin/env bash
set -euo pipefail

s=$FOO # "a b"
./bin/prep $s
./bin/setup "$secret"
```

```
  postcreate:
    - cmd: bin/prep
      args: [""]
    - cmd:
      args: []
```

## contract

1. env vars
2. cwd

## extensibility

```
{
  "hooks": {
     "postcreate": { <--- hook
        "command": "bin/setup"
        "commands": [
           "bin/setup",
        ]
        "timeout": 
     }
   }
}
```



```
{
  "hooks": {
     "postcreate": { <--- hook
        "commands": [
           "gsutil mb gs://$BUCKET",
        ]
        "timeout": 
     }
   }
}
```

os.Exec(cmd)

os.Exec("/bin/sh", "-c", "set -euo pipefail; "+cmd)